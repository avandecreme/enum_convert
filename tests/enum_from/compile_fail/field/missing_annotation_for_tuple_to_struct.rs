use enum_convert::EnumFrom;

enum Source {
    Tuple(i32, i32),
}

#[derive(EnumFrom)]
#[enum_from(Source)]
enum Target {
    #[enum_from(Source::Tuple)]
    Struct {
        // #[enum_from(Source::Tuple.1)] missing
        a: i64,
        #[enum_from(Source::Tuple.0)]
        b: i32,
    },
}

fn main() {}
