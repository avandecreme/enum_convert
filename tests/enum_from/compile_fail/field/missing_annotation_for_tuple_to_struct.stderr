error: Missing required mapping to named field for Source::Tuple
  --> tests/enum_from/compile_fail/field/missing_annotation_for_tuple_to_struct.rs:13:9
   |
13 |         a: i64,
   |         ^
